### PR TITLE
Add ability to report beatmapsets

### LINKED ISSUE #7327: Report beatmaps from beatmap page
covering reports for stolen or nsfw content. this can be handled using a similar format to user or score web reports

### PR DESCRIPTION
**Pull request by venix12**
resolves #7327

only unwanted content and beatmap stealing for now (can add more easily), also adds ``BeatmapsetMenu`` component that could be used for other things in the future as well :eyes:

### CODE REVIEW

**Comment by peppy** [comment]
I don't know where the term "Beatmap Stealing" comes from, but that shouldn't be what this report form is used for. That should be handled via a DMCA request exclusively.

**Comment by venix12** [comment]
Well, since a long time there are cases of people just taking one's beatmap and submitting it (as-is or slightly modified), it used to be handled by the moderation teams with a proper silence.

**Comment by peppy** [comment]
Let's not do that any more. It's not the correct process.

File: resources/assets/lib/beatmapsets-show/beatmapset-menu.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                key='report'
+                reportableId={beatmapset.id.toString()}
+                reportableType='beatmapset'
+                user={beatmapset.user}
+            />
+        );
+
+        return (
+            <PopupMenuPersistent>
+                {children}
```
Comment (notbakaneko) [inline resources/assets/lib/beatmapsets-show/beatmapset-menu.tsx:30]: Separate variable is kind of unnecessary with only a single item with no other significant logic.

File: resources/assets/coffee/react/beatmapset-page/header.coffee [hunk 2]
```diff
@@ -184,6 +185,11 @@ export class Header extends React.Component
 
             @renderLoginButton()
 
+            if currentUser.id?
```
Comment (notbakaneko) [inline resources/assets/coffee/react/beatmapset-page/header.coffee:188]: Also shouldn't show the option to the beatmapset owner.

**Review by notbakaneko — CHANGES_REQUESTED** [review]
This looks kind of odd, to be honest
image

The dots are also just floating there with no feedback at all

**Comment by venix12** [comment]
what would you suggest instead of those 3 dots, perhaps a button with them inside?

**Comment by peppy** [comment]
@arflyte can you check on https://github.com/ppy/osu-web/pull/7480#pullrequestreview-677168712 and propose a quick design for this?

**Comment by arflyte** [comment]
On it. Gimme a moment.

**Comment by arflyte** [comment]
I think we can use the same design as the fav button but with the exclamation triangle. Something similar can be easily added for the new beatmap info.

**Comment by venix12** [comment]
well, according to https://github.com/ppy/osu-web/issues/7327#issuecomment-818384764 it should be hidden under a dropdown menu

**Comment by notbakaneko** [comment]
I think the idea is that the button shouldn't be prominent and should be more discreet and subdued in appearance like on the use profile page? Thoughts, @arflyte?

**Comment by peppy** [comment]
@arflyte still waiting for a response from you on this one. as per the previous comments, we don't want this button to be prominent because it is seldom used.

**Comment by arflyte** [comment]
Missed this. Haven't give this much thought yet, as I was focused on the newer version of the page. We can use the style that we used on the profile page. Keep the ellipsis and put them in a transparent black circle so that it doesn't feel like it's floating. Use the same height as the buttons. Make the circle darker on hover.

File: resources/assets/lib/beatmapsets-show/beatmapset-menu.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import { BeatmapsetJson } from 'beatmapsets/beatmapset-json';
+import { PopupMenuPersistent } from 'popup-menu-persistent';
+import * as React from 'react';
+import { ReportReportable } from 'report-reportable';
+
+interface Props {
+  beatmapset: BeatmapsetJson;
+}
+
+export const BeatmapsetMenu = (props: Props) => (
```
Comment (notbakaneko) [inline resources/assets/lib/beatmapsets-show/beatmapset-menu.tsx:13]: `export default function`

**Review by notbakaneko — CHANGES_REQUESTED** [review]

**Review by notbakaneko — APPROVED** [review]

**Pull request merged**